File: app/src/main/java/com/jakubwyc/mysecretvault/mapper/TreasureMapper.java
package com.jakubwyc.mysecretvault.mapper;


import com.jakubwyc.mysecretvault.model.Treasure;
import com.jakubwyc.mysecretvault.repository.RealmTreasure;

public class TreasureMapper implements Mapper<Treasure, RealmTreasure> {

    @Override
    public RealmTreasure map(Treasure data) {
        RealmTreasure realmTreasure = new RealmTreasure();
        realmTreasure.text = data.getText();
        realmTreasure.date = data.getDate();
        return realmTreasure;
    }
}

File: app/src/main/kotlin/com/jakubwyc/mysecretvault/model/mapper/TreasureMapper.kt
package com.jakubwyc.mysecretvault.model.mapper


import com.jakubwyc.mysecretvault.model.Treasure
import com.jakubwyc.mysecretvault.repository.RealmTreasure

class TreasureMapper : Mapper<Treasure, RealmTreasure> {

    override fun map(data: Treasure): RealmTreasure {
        val realmTreasure = RealmTreasure()
        realmTreasure.text = data.text
        realmTreasure.date = data.date
        return realmTreasure
    }
}
